lib/portage/util/whirlpool.py

```diff
@@ -39,7 +39,7 @@ class Whirlpool:
 
     def __init__(self, arg=None):
```

Comment:
How about:

```suggestion
    def __init__(self, arg=b""):
```

and then you wouldn't have to check `arg` at all, the other condition will catch it.

Reply:
Works for me.